sdk/servicebus/service-bus/src/constructorHelpers.ts

```diff
@@ -116,13 +117,11 @@ export function getEntityNameFromConnectionString(connectionString: string): str
  * @internal
  * @ignore
  */
-export function convertToInternalReceiveMode(
-  receiveMode: "peekLock" | "receiveAndDelete"
-): ReceiveMode {
+export function convertToInternalReceiveMode(receiveMode: ReceiveMode): InternalReceiveMode {
   switch (receiveMode) {
     case "peekLock":
-      return ReceiveMode.peekLock;
+      return InternalReceiveMode.peekLock;
     case "receiveAndDelete":
-      return ReceiveMode.receiveAndDelete;
+      return InternalReceiveMode.receiveAndDelete;
```

Comment:
We can remove the `InternalReceiveMode` enum completely in favor of the new `ReceiveMode`.
Can be done in a separate PR.

Reply:
@ramya-rao-a